en/editor/publish/build-guide.md

```diff
@@ -0,0 +1,186 @@
+    // "bba0...@6c48a": ["test/right/texture", 1, 1]
+    scenes: Array<{url: string, uuid: string}>; // The array of scenes information involved in the run
+    scriptPackages: Array<{moduleId: string, file: string}>; // Script message array
+    jsList: string[]; // Script plugin array
+    moduleIds: string[]; // Information on all user script components
+    packedAssets: Record<string, IUuid[] | number[]>; // json grouping information
+    md5AssetsMap: { [index: string]: Array<string | number> }; // It is not available until md5Cache is checked, and the array is stored in the format of [uuid_1, md5_1, uuid_2, md5_2, ...]. If uuid_1 is a simple number, it means that the uuid index in the uuids array is stored.
+    uuids: string[]; // Arrays of uuid, only takes effect in release mode
+    assetTypes?: string[]; // Arrays of resource types, only takes effect in release mode
+    subpackages?: Record<IUuid, IPackageInfo>; // Subpackage resource information
+    renderPipeline: string; // renderPipeline information
+}
+```
+
+The structure here only lists the settings structure under the common process, and actually adds configurations as needed when packaging for different platforms.
+
+### Compression and writing of `settings uuid`
+
+During the resource packaging process, the `uuid` of the resources involved in the build of the resource is continuously collected, and then organized into `setting.js`. `setting.js` will be written to the `build/src` directory generated after the build, and the `uuid` in the file will be compressed or not, depending on whether it is in debug mode or not. Organize all used `uuid` and store the `uuid` that appear more than twice in the `uuids` array, and replace them with indexes. All `assetType` that appear more than twice are also stored in the `assetTypes` array, and replace them with indexes.
+
+#### Build resources
+
+The packaging of resources in this step does not include scripts, because scripts are packaged as special files. The editor will first summarize the scene currently involved in the build and all resources in the `resources` directory. Each resource is packaged through the engine's deserialization to find the dependent resource and recursion to pack the resources. The entire project's scripting environment is configured before being deserialized, that is, all non-plugin project scripts are loaded. Because whether the script loads correctly or not directly affects the deserialization, failure to load because the script is not written legally will directly result in build failure. If the dependent resource is lost in the deserialization process, a warning is issued, but the build continues nonetheless. The warning here does not mean that the problem does not need to be resolved, and if the resource loss is not resolved, it is difficult to guarantee that the problem will not occur after the build.
```

Comment:
please refer this Chinese document: https://github.com/cocos-creator/docs-3d/blob/master/zh/editor/publish/build-guide.md#%E6%9E%84%E5%BB%BA%E8%B5%84%E6%BA%90

Reply:
#### Build assets

At this stage, the editor will summarize **scenes assets that checked in build panel and all assets in the `resources` directory. All assets will be deserialized to find out the dependent assets in deep**.Before deserialization, editor will load all the scripts（expect plugin scripts) in the project,if the script is written illegally and fails to load, it will directly fail to build.If any dependent asset is missing during the deserialization process, a warning info will be print, but editor will continue to build.When warning info is printed,we recommend you to read it and try to resolve it. ** If the assets is lost and this problem not resolved, it may cause some unexpect errors after build**.During the packaging process, the assets will be re-compressed and serialized after deserialization to reduce the package size.All serialized files will be sorted into deferent JSON groups according to the build configuration parameters.

Reply:
已改